### ISSUE TITLE
qwen3-vl系列的模型对grouding的坐标进行了1000倍的缩放，那以前的训练数据是不是也要修正呢？

### ISSUE THREAD

**Issue by Hormoney** [QwenLM/Qwen3-VL#1623]
cookbook中提到了绝对位置缩放的内容
 # Convert normalized coordinates to absolute coordinates
      abs_y1 = int(bounding_box["bbox_2d"][1] / 1000 * height)
      abs_x1 = int(bounding_box["bbox_2d"][0] / 1000 * width)
      abs_y2 = int(bounding_box["bbox_2d"][3] / 1000 * height)
      abs_x2 = int(bounding_box["bbox_2d"][2] / 1000 * width)

那么我之前训qwen2.5-vl的训练数据，是不是需要把label的box坐标也进行这样的缩放啊

**Comment by JJJYmmm** [comment]
是的，需要量化到0-1000

**Comment by Hormoney** [comment]
> 是的，需要计算到0-1000

大佬，我之前的坐标是28的倍数，现在是不是需要先转换成32 的倍数然后在进行0-1000的缩放呢
Specify exact dimensions: Directly set resized_height and resized_width. These values will be rounded to the nearest multiple of 32 (32 for Qwen3VL, 28 for Qwen2.5VL)

**Comment by JJJYmmm** [comment]
不用，只需要保证28倍数的坐标量化时使用的img_w, img_h是经过smart size后的就行

**Comment by Hormoney** [comment]
> 不用，只需要保证28倍数的坐标量化时使用的img_w, img_h是经过smart size后的就行

感谢大佬

**Comment by qianyue76** [comment]
> 是的，需要量化到0-1000

为啥qwen3-vl又改回了相对坐标

**Comment by gaolongxi** [comment]
> 不用，只需要保证28倍数的坐标量化时使用的img_w, img_h是经过smart size后的就行

如果基于Qwen3VL训练Grounding任务，图片分辨率比较高，构造数据，训练时的sh脚本，以及在benchmark上面评测时，是否都应该保证这三处的min_pixels, max_pixels是一致的？目前看到的几处默认参数设置并不一致，需要自己修改吗？
1. 训练脚本给出min_pixels=784, max_pixels=50176：https://github.com/QwenLM/Qwen3-VL/blob/main/qwen-vl-finetune/scripts/sft_qwen3_4b.sh
2. 这里依然是qwen2.5vl config的默认值：https://github.com/QwenLM/Qwen3-VL/blob/main/qwen-vl-finetune/tools/process_bbox.ipynb
3. usage示例里面给出min_pixels=16\*28\*28, max_pixels=576\*28\*28, https://github.com/QwenLM/Qwen3-VL/blob/main/qwen-vl-finetune/README.md
另外，对于高分辨率的grounding，max_pixels是否设置的越大越好，构造数据，训练时的sh脚本，以及在benchmark上面评测都使用min_pixels=16\*28\*28, max_pixels=576\*28\*28效果会是最佳吗？

**Comment by Shame-fight** [comment]
> 是的，需要量化到0-1000

大佬，请问微调grounding是直接将原始图像坐标缩放到0-1000吗？
我在使用llama factory框架训练，训练配置文件中的image_max_pixels是否需要填写。
`### model
model_name_or_path: /root/autodl-tmp/nanshu/models/qwen3vl-8b
quantization_bit: 4  # choices: [8 (bnb/hqq/eetq), 4 (bnb/hqq), 3 (hqq), 2 (hqq)]
quantization_method: bnb  # choices: [bnb, hqq, eetq]
image_max_pixels: 802816 #262144
#image_min_pixels: 250880


#trust_remote_code: true

### method
stage: sft
do_train: true
finetuning_type: lora
lora_rank: 8
lora_target: all
#deepspeed: examples/deepspeed/ds_z2_config.json  # choices: [ds_z0_config.json, ds_z2_config.json, ds_z3_config.json]

### dataset
dataset: dianwang_1,dianwang_3,gczy # dianwang_labelme #dianwnag_test,dianwnag_test2,dianwnag_test3,dianwnag_test4,dianwnag_test5,dianwnag_test6,dianwnag_test7,dianwnag_test8,dianwnag_mj2
template: qwen2_vl
cutoff_len: 4096 #2048  4096
max_samples: 10000
overwrite_cache: true
preprocessing_num_workers: 4
dataloader_num_workers: 4

### output
output_dir: saves/qwen2_5vl-8b/qlora/sft_deepspeed/20251016_01/
logging_steps: 10
save_steps: 100
plot_loss: true
overwrite_output_dir: true
save_only_model: false
report_to: none  # choices: [none, wandb, tensorboard, swanlab, mlflow]

### train
per_device_train_batch_size: 10
gradient_accumulation_steps: 2
learning_rate: 1.0e-4
num_train_epochs: 7
lr_scheduler_type: cosine
warmup_ratio: 0.1
bf16: true
ddp_timeout: 180000000
resume_from_checkpoint: null`

**Comment by kaidi00** [comment]
> > 不用，只需要保证28倍数的坐标量化时使用的img_w, img_h是经过smart size后的就行
> 
> 感谢大佬

什么意思，大佬能讲讲吗

**Comment by Moyhub** [comment]
> 不用，只需要保证28倍数的坐标量化时使用的img_w, img_h是经过smart size后的就行

请问一下为什么不需要呢？假设我的原始图是1300*1300，在qwen2.5-vl中，图片会被处理成1288，1288（28的倍数），我对应的grounding坐标也是基于此图片大小进行了变换，成为训练数据A（原始的grounding坐标是基于原始图片大小的）。而在qwen3-vl中，smart_resize会将图片大小变化成1312,1312。那么我的grounding坐标是不是应该先基于1312,1312的大小进行变换后，在缩放到0-1000以内。这样的话不就是要将训练数据A的28倍数坐标转化为32倍数坐标，再进行缩放吗？

补充：如果这里的smart_resize用的是qwen2.5-vl的，即factor=28。那是直接将当前的28倍数的坐标点，直接除以当前图的长和宽即可。因为是相对的点，因此只要当前点是对应于当前图大小的量化即可。按我上文的理解，如果将28倍数的图转化为32倍数的图，h,w都会变换，即w*x_ratio, h*y_ratio。 对应地，28倍数下的图的坐标，x,y 也会相应的变化，x*x_ratio, y*y_ratio. 量化到1000式，x*ratio/w*ratio * 1000 = x/w * 1000，即直接将28倍数的坐标对应于28倍数下的图进行规范化0-1000即可。

**Comment by JJJYmmm** [comment]
@gaolongxi 对于高分辨率grounding，min/max pixel (shortest/longest edge)设置一个比较大的范围就可以，例如400*32*32 ～ 6000*32*32（28是2.5vl的patch size，3vl是32）；另外可能对于不同任务，最优的pixel range不太一样，所以还是看具体任务，一般根据训练上下文长度统一设置就可以了；

**Comment by JJJYmmm** [comment]
@Shame-fight 可以，直接量化到0-1000就可以；image_max_pixels取决于你的训练资源

**Comment by JJJYmmm** [comment]
@kaidi00 这里说的是2.5VL的数据到3VL数据的转化，如果你的是原始grounding数据，直接转成相对坐标就可以了

**Comment by JJJYmmm** [comment]
@Moyhub 嗯，相对坐标都是一样的

**Comment by gaolongxi** [comment]
> [@gaolongxi](https://github.com/gaolongxi) 对于高分辨率grounding，min/max pixel (shortest/longest edge)设置一个比较大的范围就可以，例如400_32_32 ～ 6000_32_32（28是2.5vl的patch size，3vl是32）；另外可能对于不同任务，最优的pixel range不太一样，所以还是看具体任务，一般根据训练上下文长度统一设置就可以了；

感谢大佬！我使用Qwen2.5-VL-7B在1080*1920的mobile GUI上面SFT全参数微调Grounding任务，构造数据，训练和推理都是使用相同的min/max pixel: 3136-1003520，这好像是Qwen2.5-VL的最大限制范围，但是训练后在benchmark上面掉点非常严重，训练后几乎完全错误。但是相同的数据，在GRPO之后都有提分。SFT使用的是官方仓库提供的训练脚本，这可能是什么原因导致的？min/max范围给的太大了吗？

**Comment by JJJYmmm** [comment]
@gaolongxi 使用的是这个repo下的微调代码吗？“几乎完全错误”是指训练后的模型预测反而更差了嘛

**Comment by gaolongxi** [comment]
> [@gaolongxi](https://github.com/gaolongxi) 使用的是这个repo下的微调代码吗？“几乎完全错误”是指训练后的模型预测反而更差了嘛

感谢大佬回复！使用的是这个仓库Qwen2.5-VL时期提供的微调代码：
```
#!/bin/bash

# Distributed training configuration
NPROC_PER_NODE=8
MASTER_ADDR=${MASTER_ADDR:-"127.0.0.1"}
MASTER_PORT=${MASTER_PORT:-$(shuf -i 20001-29999 -n 1)}
NNODES=${WORLD_SIZE:-1}

# DeepSpeed configuration
deepspeed=./scripts/zero3.json

# Model configuration
llm={model_path}

# Training hyperparameters
lr=2e-7
batch_size=4
grad_accum_steps=4

# Training entry point
entry_file=qwenvl/train/train_qwen.py

# Dataset configuration (replace with public dataset names)
datasets=gui_grounding_25k

# Output configuration
run_name="gui_grounding_25k"
output_dir=./output/${run_name}

# Training arguments
args="
    --deepspeed ${deepspeed} \
    --model_name_or_path "${llm}" \
    --dataset_use ${datasets} \
    --data_flatten True \
    --tune_mm_vision False \
    --tune_mm_mlp False \
    --tune_mm_llm True \
    --bf16 \
    --output_dir ${output_dir} \
    --num_train_epochs 1 \
    --per_device_train_batch_size ${batch_size} \
    --per_device_eval_batch_size $((batch_size*2)) \
    --gradient_accumulation_steps ${grad_accum_steps} \
    --max_pixels $((56*56)) \
    --min_pixels $((1280*28*28)) \
    --eval_strategy "no" \
    --save_strategy "steps" \
    --save_steps 1000 \
    --save_total_limit 1 \
    --learning_rate ${lr} \
    --weight_decay 0 \
    --warmup_ratio 0.03 \
    --max_grad_norm 1 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --model_max_length 8192 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --run_name ${run_name} \
    --report_to wandb"

# Launch training
torchrun --nproc_per_node=${NPROC_PER_NODE} \
         --master_addr=${MASTER_ADDR} \
         --master_port=${MASTER_PORT} \
         ${entry_file} ${args}
```
我的一条训练数据如下：
```
  {
    "image": "214369.jpg",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\nPoint to the UI element most relevant to \"A button that prompts the user to update the application to the latest version.\", and output the bbox coordinates in JSON format."
      },
      {
        "from": "gpt",
        "value": "{\n\"bbox_2d\": [382, 686, 614, 771]\n}"
      }
    ]
  }
```
我的训练数据和测试的benchmark图片完全不相同，SFT训练后模型完全预测不准了，epoch=1比epoch=4稍好一些，我的训练集图片和用来评测的benchmark(ScreenSpot-v2, -Pro) 图片完全不同，所以没提升可以理解，现在的情况是准确率趋近0了，有点奇怪。同样的训练数据，使用GRPO算法训练，就有提升。不知道是否只是算法的问题，还是我的训练数据构造或者超参数设置有问题呢？训练数据的groundtruth是使用smart_resize按照Qwen2.5-VL的逻辑处理过的，代码如下所示。
```
def parse_bbox_qwenvl2_5(bbox_str, height, width):
    """
    把"[[x1, y1], [x2, y2]]"解析成[x1, y1, x2, y2]的列表。
    """
    resized_h, resized_w = smart_resize(height, width, factor=28, min_pixels=56 * 56, max_pixels=14 * 14 * 4 * 1280)
    arr = ast.literal_eval(bbox_str)
    (x1, y1), (x2, y2) = arr
    resized_x1 = int(x1 / width * resized_w)
    resized_y1 = int(y1 / height * resized_h)
    resized_x2 = int(x2 / width * resized_w)
    resized_y2 = int(y2 / height * resized_h)
    return [resized_x1, resized_y1, resized_x2, resized_y2]
```

**Comment by AdamChrist** [comment]
> [@Moyhub](https://github.com/Moyhub) 嗯，相对坐标都是一样的

请问，如果我想使用`sionic-ai/refcocog_object_detection` 的数据集来微调qwen3vl-8b，是不是应该先使用 `process_vision_info ` 来处理图片，并且把图片中的bbox坐标也做相对的处理呢？

**Comment by huangl22** [comment]
> 是的，需要量化到0-1000

您好！请问基于什么量化到0-1000呀？是基于当前标注的bbox的最大值进行量化吗？还是基于当前照片的width和height进行量化？x1 = int(x1 / 1000 * width)
        y1 = int(y1 / 1000 * height)
        x2 = int(x2 / 1000 * width)
        y2 = int(y2 / 1000 * height)
